packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -26,15 +26,20 @@ import { store as editorStore } from '../../store';
 import { lock, unlock } from '../../lock-unlock';
 import { useGlobalStylesContext } from '../global-styles-provider';
 
-const EMPTY_BLOCKS_LIST = [];
 const EMPTY_OBJECT = {};
 
 function __experimentalReusableBlocksSelect( select ) {
-	return (
-		select( coreStore ).getEntityRecords( 'postType', 'wp_block', {
-			per_page: -1,
-		} ) ?? EMPTY_BLOCKS_LIST
-	);
+	const { getEntityRecords, hasFinishedResolution } = select( coreStore );
+	const reusableBlocks = getEntityRecords( 'postType', 'wp_block', {
+		per_page: -1,
+	} );
+	return hasFinishedResolution( 'getEntityRecords', [
+		'postType',
+		'wp_block',
+		{ per_page: -1 },
+	] )
+		? reusableBlocks
+		: undefined;
```

Comment:
I don't understand this `hasFinishedResolution` technique. If there are valid `reusableBlocks` in state, but resolution hasn't finished, then the resolution must have been invalidated. And the state contains stale data that we don't want. So we add this condition to avoid getting the stale data. Is that right?

Reply:
I'm not sure I'm following. Could you elaborate? Sorry

Reply:
I'm more or less asking why we call `hasFinishedResolution` instead of just always returning `reusableBlocks` from the selector.